android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java

```diff
@@ -0,0 +1,110 @@
+import org.chromium.ui.modaldialog.ModalDialogProperties;
+import org.chromium.ui.modaldialog.ModalDialogProperties.ButtonStyles;
+import org.chromium.ui.modaldialog.SimpleModalDialogController;
+import org.chromium.ui.modelutil.PropertyModel;
+import androidx.annotation.NonNull;
+import org.chromium.chrome.browser.crypto_wallet.util.WalletConstants;
+import org.chromium.chrome.browser.app.BraveActivity;
+import org.chromium.components.browser_ui.modaldialog.ModalDialogView;
+import android.annotation.SuppressLint;
+import android.widget.Button;
+import org.chromium.chrome.R;
+
+
+
+public class BraveNotificationPermissionRationaleDialogController extends NotificationPermissionRationaleDialogController {
+
+    private final ModalDialogManager mModalDialogManager;
+    private final Context mContext;
+
+
+	public BraveNotificationPermissionRationaleDialogController(
+            Context context, ModalDialogManager modalDialogManager) {
+        super(context, modalDialogManager);
+        mModalDialogManager = modalDialogManager;
+        mContext = context;
+    }
+
+    @Override
+    public void showRationaleUi(Callback<Boolean> rationaleCallback) {
+        LayoutInflater inflater = LayoutInflater.from(mContext);
+        Resources resources = mContext.getResources();
+
+        View dialogView = inflater.inflate(R.layout.brave_notification_permission_rationale_dialog,
+                /* root= */ null);
+        TextView titleView = dialogView.findViewById(R.id.notification_permission_rationale_title);
+        TextView descriptionView =
+                dialogView.findViewById(R.id.notification_permission_rationale_message);
+
+        PropertyModel.Builder dialogModelBuilder =
+                new PropertyModel.Builder(ModalDialogProperties.ALL_KEYS)
+                        .with(ModalDialogProperties.CONTROLLER,
+                                new SimpleModalDialogController(mModalDialogManager,
+                                        wrapDialogDismissalCallback(rationaleCallback)))
+                        .with(ModalDialogProperties.CUSTOM_VIEW, dialogView);
+
+        PropertyModel dialogModel = dialogModelBuilder.build();
+        
+        //continue button
+        Button btnContinue = dialogView.findViewById(R.id.notification_continue_button);
+        btnContinue.setOnClickListener((v) -> {
+            mModalDialogManager.dismissDialog(
+                    dialogModel, DialogDismissalCause.POSITIVE_BUTTON_CLICKED);
+        });
+
+        //now Button
+        Button btnNotNow = dialogView.findViewById(R.id.notification_not_now_button);
+        btnNotNow.setOnClickListener((v) -> {
+            mModalDialogManager.dismissDialog(
+                    dialogModel, DialogDismissalCause.NEGATIVE_BUTTON_CLICKED);
+        });
+
+        mModalDialogManager.showDialog(dialogModel, ModalDialogType.APP);
+    }
+
+    private Callback<Integer> wrapDialogDismissalCallback(Callback<Boolean> rationaleCallback) {
```

Comment:
This seems to be 1 to 1 copy of the parent's implementation. If so, we should use parent's call directly. See `destroyPropertiesForStream` as an example.

Reply:
even 1 to 1 copy but it's private to that class.
and public method showRationaleUi it's changed all , so no need to call super

Reply:
We can make it public in bytecode and I don't really see any advantage of having a copy of it. Are we planing to extend it?